File: 백준/Silver/28278. 스택 2/스택 2.py
from collections import deque
import sys

N = int(sys.stdin.readline())
stack = deque()
for i in range(N):
    order = sys.stdin.readline().split()

    if order[0] == '1':
        stack.append(order[1])

    elif order[0] == '2':
        if len(stack) != 0:
            print(stack.pop())
        else:
            print(-1)

    elif order[0] == '3':
        print(len(stack))

    elif order[0] == '4':
        if stack:
            print(0)
        else:
            print(1)

    else:
        if stack:
            print(stack[-1])
        else:
            print(-1)

File: 백준/Silver/28278. 스택 2/스택 2.cc
#include <iostream>
#include <stack>

using namespace std;

int main() {
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	int N;
	cin >> N;

	stack<int>s;
	int order;
	
	for (int i = 0;i < N;i++) {
		cin >> order;
		switch (order) {
		case 1:
			int num;
			cin >> num;
			s.push(num);
			break;
		case 2:
			if (!s.empty()) {
				cout << s.top() << '\n';
				s.pop();
			}
			else {
				cout << -1 << '\n';
			}break;
		case 3:
			cout << s.size() << '\n';
			break;
		case 4:
			if (!s.empty()) {
				cout << 0 << '\n';
			}
			else {
				cout << 1 << '\n';
			}break;
		case 5:
			if(!s.empty()){
				cout << s.top() << '\n';
			}
			else {
				cout << -1 << '\n';
			}break;
		}

	}

	return 0;
}